cli/cli/helpers/grafloki/kubernetes_grafloki.go

```diff
@@ -0,0 +1,477 @@
+				StorageOS:             nil,
+				CSI:                   nil,
+				Ephemeral:             nil,
+			},
+		}},
+		&apiv1.Affinity{
+			NodeAffinity:    nil,
+			PodAffinity:     nil,
+			PodAntiAffinity: nil,
+		})
+	shouldRemoveGrafanaDeployment := true
+	defer func() {
+		if shouldRemoveGrafanaDeployment {
+			if err := k8sManager.RemoveDeployment(ctx, graflokiNamespace, grafanaDeployment); err != nil {
+				logrus.Warnf("Attempted to remove Loki deployment after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Loki deployment with name: %v", lokiDeployment.Name)
+			}
+		}
+	}()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Grafana deployment.")
+	}
+
+	grafanaService, err := k8sManager.CreateService(ctx,
+		graflokiNamespace,
+		grafanaServiceName,
+		map[string]string{}, // empty labels
+		nil,                 // empty annotations
+		grafanaLabels,       // match grafana deployment pod labels
+		apiv1.ServiceTypeNodePort,
+		[]apiv1.ServicePort{{
+			Name:        "grafana-dashboard",
+			Port:        grafanaNodePort,
+			TargetPort:  intstr.FromInt(grafanaPort),
+			Protocol:    apiv1.ProtocolTCP,
+			NodePort:    grafanaNodePort,
+			AppProtocol: &httpApplicationProtocol,
+		}})
+	shouldRemoveGrafanaService := true
+	defer func() {
+		if shouldRemoveGrafanaService {
+			if err := k8sManager.RemoveService(ctx, grafanaService); err != nil {
+				logrus.Warnf("Attempted to remove Grafana service after an error occurred creating it but an error occurred removing it.")
+				logrus.Warnf("Manually remove Grafana service with name: %v", grafanaService.Name)
+			}
+		}
+	}()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred creating Grafana service")
+	}
+
+	shouldRemoveLokiDeployment = false
+	shouldRemoveGrafanaConfigMap = false
+	shouldRemoveGrafanaDeployment = false
+	shouldRemoveGrafanaService = false
+	shouldRemoveNamespace = false
+	shouldRemoveLokiService = false
+	return lokiHost, nil
+}
+
+func checkGrafanaAndLokiDeploymentExistence(ctx context.Context, k8sManager *kubernetes_manager.KubernetesManager) (bool, string) {
+	existsLoki := false
+	existsGrafana := false
+	var lokiHost string
+	lokiDeployment, err := k8sManager.GetDeployment(ctx, graflokiNamespace, lokiDeploymentName)
+	if err == nil && lokiDeployment != nil {
+		existsLoki = true
+		lokiHost = getLokiHost(lokiServiceName, graflokiNamespace, lokiNodePort)
+	}
+
+	grafanaDeployment, err := k8sManager.GetDeployment(ctx, graflokiNamespace, grafanaDeploymentName)
+	if err == nil && grafanaDeployment != nil {
+		existsGrafana = true
+	}
+
+	return existsLoki && existsGrafana, lokiHost
+}
+
+func StopGrafLokiInKubernetes(ctx context.Context) error {
+	k8sManager, err := getKubernetesManager()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred getting Kubernetes Manager.")
+	}
+	ns, err := k8sManager.GetNamespace(ctx, graflokiNamespace)
```

Comment:
Same comment about non-common abbreviations (i.e. not API, etc.). Feel free to totally ignore though; I just personally find it makes the code harder to parse